datumaro/plugins/open_images_format.py

```diff
@@ -0,0 +1,155 @@
+    FULL_IMAGE_DESCRIPTION_NAME = 'image_ids_and_rotation.csv'
+    SUBSET_IMAGE_DESCRIPTION_PATTERNS = (
+        '*-images-with-rotation.csv',
+        '*-images-with-labels-with-rotation.csv',
+    )
+
+class OpenImagesExtractor(Extractor):
+    def __init__(self, path):
+        if not osp.isdir(path):
+            raise FileNotFoundError("Can't read dataset directory '%s'" % path)
+
+        super().__init__()
+
+        self._dataset_dir = path
+        self._categories = {}
+        self._items = []
+
+        self._load_categories()
+        self._load_items()
+
+    def __iter__(self):
+        return iter(self._items)
+
+    def categories(self):
+        return self._categories
+
+    @contextlib.contextmanager
+    def _open_csv_annotation(self, file_name):
+        absolute_path = osp.join(self._dataset_dir, _OpenImagesPath.ANNOTATIONS_DIR, file_name)
+
+        with open(absolute_path, 'r', newline='') as f:
+            yield csv.DictReader(f)
+
+    def _glob_annotations(self, pattern):
+        return glob.glob(osp.join(
+            glob.escape(self._dataset_dir), _OpenImagesPath.ANNOTATIONS_DIR, pattern))
+
+    def _load_categories(self):
+        label_categories = LabelCategories()
+
+        with self._open_csv_annotation('oidv6-class-descriptions.csv') as class_description_reader:
+            for class_description in class_description_reader:
+                label_categories.add(class_description['LabelName'])
+
+        self._categories[AnnotationType.label] = label_categories
+
+        self._load_label_category_parents()
+
+    def _load_label_category_parents(self):
+        label_categories = self._categories[AnnotationType.label]
+
+        hierarchy_path = osp.join(
+            self._dataset_dir, _OpenImagesPath.ANNOTATIONS_DIR, 'bbox_labels_600_hierarchy.json')
+
+        with open(hierarchy_path, 'rb') as hierarchy_file:
+            root_node = json.load(hierarchy_file)
+
+        def set_parents_from_node(node, category):
+            for child_node in node.get('Subcategory', []):
+                _, child_category = label_categories.find(child_node['LabelName'])
+
+                if category is not None and child_category is not None:
+                    child_category.parent = category.name
+
+                set_parents_from_node(child_node, child_category)
+
+        _, root_category = label_categories.find(root_node['LabelName'])
+        set_parents_from_node(root_node, root_category)
+
+    def _load_items(self):
+        items_by_id = {}
+
+        def load_from(path):
+            with self._open_csv_annotation(path) as image_reader:
+                for image_description in image_reader:
+                    image_id = image_description['ImageID']
+                    assert _RE_IMAGE_ID.fullmatch(image_id)
+                    assert image_id not in items_by_id
+
+                    subset = image_description['Subset']
+                    assert subset in {'test', 'train', 'validation'}
```

Comment:
We don't limit subset names to these 3, unless format can't represent others (like YOLO, where we have only `train` and `valid`).

Reply:
OK. However, we can't just allow subset names to be _anything_, because they're used as path components to locate image files (e.g. we shouldn't allow `../../../..` to be used as a subset name, since then we would end up accessing data from outside the dataset). Is there some pre-existing functionality in Datumaro to check if a name is path-safe?

Reply:
No, there is no such functionality, but disallowing slashes looks like a reasonable limitation.

Reply:
I loosened up the subset name validation rules (and replaced the assert with a custom exception).